File: backend/src/main/java/eu/gillespie/bscthesis/request/ConditionalDefinition.java
package eu.gillespie.bscthesis.request;

import com.google.gson.annotations.SerializedName;
import eu.gillespie.bscthesis.smt.v20.model.interfaces.SmtV20AssertableExpression;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@ToString
@Getter
public class ConditionalDefinition {
    @SerializedName("condition")
    StatementTreeVertex condition;

    @SerializedName("then")
    StatementTreeVertex then;

}

File: backend/src/main/java/eu/gillespie/bscthesis/request/ConditionalDefinition.kt
package eu.gillespie.bscthesis.request;

import lombok.Getter;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@ToString
@Getter
@RequiredArgsConstructor
public class ConditionalDefinition {
    @NonNull StatementTreeVertex condition;
    @NonNull StatementTreeVertex then;

}
